capstone/files/enrichment/virustotal.py

```diff
@@ -0,0 +1,187 @@
+# September 8, 2020.
+#
+# This program is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+# Wazuh Inc <support@wazuh.com>
+# Google LLC
+
+import json
+import logging
+import os
+import sys
+from socket import socket, AF_UNIX, SOCK_DGRAM
+
+try:
+    import vt
+except ImportError as e:
+    print("No module 'vt' found."
+          "Follow guide here https://virustotal.github.io/vt-py/howtoinstall.html")
+    print("WARNING! Do not install this library using pip!"
+          "Currently released version can break this script.")
+    sys.exit(1)
+
+# ossec.conf configuration:
+#  <integration>
+#      <name>virustotal</name>
+#      <api_key>api_key_here</api_key>
+#      <group>syscheck</group>
+#      <alert_format>json</alert_format>
+#  </integration>
+
+DEBUG = False
+PWD = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+
+# Set paths
+LOG_FILE_PATH = f'{PWD}/logs/integrations.log'
+SOCKET_ADDR = f'{PWD}/queue/ossec/queue'
+
+# Set logger
+logging.basicConfig(filename=f'{PWD}/logs/integrations.log',
+                    filemode='a',
+                    format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
+                    datefmt='%H:%M:%S',
+                    level=logging.INFO)
+
+
+def main(args):
+    """Main function for enrichment flow.
+
+    @param args: System args [file location, api key].
+    """
+    logging.info("# Starting")
+
+    # Read args
+    alert_file_location = args[1]
+    apikey = args[2]
+
+    logging.info("# API Key")
+    logging.info(apikey)
+
+    logging.info("# File location")
+    logging.info(alert_file_location)
+
+    json_alert = {}
+
+    # Load alert. Parse JSON object.
+    with open(alert_file_location) as alert_file:
+        json_alert = json.load(alert_file)
+
+    logging.info("# Processing alert")
+    logging.info(json_alert)
+
+    # Request VirusTotal info
+    msg = request_virustotal_info(json_alert, apikey)
+
+    # If positive match, send event to Wazuh Manager
+    if msg:
+        send_event(msg, json_alert["agent"])
+
+
+def get_file_info(hash, apikey):
+    """A function that gets file information from VirusTotal.
+
+    @param hash: MD5 hash of the file that is sent to VirusTotal.
+    @param apikey: VirusTotal API key.
+    @return: Instance of vt.Object that contains information about about requested file.
+    """
+    vt_client = vt.Client(apikey)
+
+    file_info = vt_client.get_object(f"/files/{hash}")
+
+    if file_info is not None:
+        return file_info
+    else:
+        alert_output = {}
+        alert_output["virustotal"] = {}
+        alert_output["integration"] = "virustotal"
+
+        logging.error("# Error when conecting VirusTotal API")
+        alert_output["virustotal"]["description"] = "Error: API request fail"
+        send_event(alert_output)
+        exit(0)
```

Comment:
It's not obvious that `get_file_info` can exit program on fail. Mark it on docstring or change to raise/return None